include/swift/Sema/CSFix.h

```diff
@@ -2023,11 +2023,13 @@ class AllowInvalidRefInKeyPath final : public ConstraintFix {
   } Kind;
 
   ValueDecl *Member;
+  Type BaseType;
 
   AllowInvalidRefInKeyPath(ConstraintSystem &cs, RefKind kind,
-                           ValueDecl *member, ConstraintLocator *locator)
+                           ValueDecl *member, ConstraintLocator *locator,
+                           Type baseType = Type())
```

Comment:
Let put `baseType` before `RefKind` and make it required always since components should always have a base type.